src/NServiceBus.Core/Persistence/InMemory/TimeoutPersister/InMemoryTimeoutPersister.cs

```diff
@@ -105,5 +106,16 @@ public void RemoveTimeoutBy(Guid sagaId)
                 readerWriterLock.ExitWriteLock();
             }
         }
+
+        public TimeoutData Peek(string timeoutId)
+        {
+            return storage.SingleOrDefault(t => t.Id == timeoutId);
```

Comment:
so assign timeout data to variable, exit lock and then return value?
Not sure if that matters that much for inmemory one, it's only for tests and dev.